examples/streaming/README.md

```diff
@@ -0,0 +1,164 @@
+
+## Snap Plugin Go Library: Stream Collector Plugin Example
+Here you will find an example plugin that covers the basics for writing a stream collector plugin.
+
+## Plugin Naming, Files, and Directory
+For your stream collector plugin, create a new repository and name your plugin project using the following format:
+
+>snap-plugin-[plugin-type]-[plugin-name]
+
+For example: 
+>snap-plugin-collector-rand
+
+
+Example files and directory structure:  
+```
+snap-plugin-[plugin-type]-[plugin-name]
+ |--[plugin-name]
+  |--[plugin-name].go  
+  |--[plugin-name]_test.go  
+ |--main.go
+```
+
+For example:
+```
+snap-plugin-collector-rand
+ |--rand
+  |--rand.go  
+  |--rand_test.go  
+ |--main.go
+```
+
+* The [plugin-name] folder (for example `rand`) will include all files to implement the appropriate interface methods
+* Your [plugin-name] folder  will also include your test files.
+
+
+
+## Interface Methods
+
+In order to write a plugin for Snap, it is necessary to define a few methods to satisfy the appropriate interfaces. These interfaces must be defined for a collector plugin:
```

Comment:
In 
>In order to write a plugin for snap..

Is it intended that these interface methods can be applied for all snap plugins and not just the streaming plugins?